src/main/java/com/xwiki/authentication/saml/SamlAuthenticator.java

```diff
@@ -0,0 +1,362 @@
+        this.groupManager = groupManager;;
+        this.authConfig = authConfig;
+        this.currentMixedDocumentReferenceResolver = currentMixedDocumentReferenceResolver;
+        this.compactStringEntityReferenceSerializer = compactStringEntityReferenceSerializer;
+        this.loginAuthFactory = loginAuthFactory;
+        this.samlSettings = buildSamlSettings();
+    }
+
+    private Saml2Settings buildSamlSettings() {
+        final Properties settings = new Properties();
+        settings.put("onelogin.saml2.strict", true);
+        settings.put("onelogin.saml2.sp.entityid", authConfig.entityId);
+        settings.put("onelogin.saml2.sp.assertion_consumer_service.url", authConfig.assertionConsumerServiceUrl);
+        settings.put("onelogin.saml2.sp.nameidformat", authConfig.nameIdFormat);
+        settings.put("onelogin.saml2.idp.entityid", authConfig.idpEntityId);
+        settings.put("onelogin.saml2.idp.single_sign_on_service.url", authConfig.idpSingleSignOnUrl);
+        settings.put("onelogin.saml2.idp.x509cert", authConfig.x509Certificate);
+        return new SettingsBuilder().fromProperties(settings).build();
+    }
+
+    public XWikiUser checkAuth(XWikiContext context,
+                               SupplierWithException<XWikiUser, XWikiException> defaultAuthHandler)
+            throws XWikiException {
+        final XWikiRequest request = context.getRequest();
+        final XWikiResponse response = context.getResponse();
+
+        final String samlUserName = getSAMLAuthenticatedUserFromSession(context);
+        if (samlUserName != null) {
+            LOG.debug("Found authenticated user [{}]", samlUserName);
+            return new XWikiUser(getUserReferenceForName(samlUserName), context.isMainWiki());
+        }
+        final String samlResponse = request.getParameter("SAMLResponse");
+        final boolean hasSamlResponse = samlResponse != null;
+        if (!hasSamlResponse) {
+            final String action = context.getAction();
+            if (actionsAllowedByUnauthenticatedUsers().contains(action))
+                return defaultAuthHandler.execute();
+
+            if (context.getRequest().getCookie("username") != null)
+                return defaultAuthHandler.execute();
+        }
+
+        try {
+            if (LOG.isDebugEnabled())
+                LOG.debug("SAML 2.0 Authentication redirection started");
+
+            final Auth auth = loginAuthFactory.produce(this.samlSettings, request, response);
+
+            if (!hasSamlResponse) {
+                auth.login(XWiki.getRequestURL(request).toString());
+                return null;
+            }
+
+            auth.processResponse();
+
+            if (auth.isAuthenticated()) {
+                final DocumentReference userReference = findOrCreateUser(context, auth);
+                if (userReference != null) {
+                    LOG.debug("Found authentication of user [{}]", auth.getNameId());
```

Comment:
```suggestion
                    LOG.debug("Found authentication for user [{}]", auth.getNameId());
```

Also, is this really Debug?